code/datums/exonet.dm

```diff
@@ -0,0 +1,226 @@
+GLOBAL_LIST_EMPTY(exonets)
+
+/datum/exonet
+	var/ennid										// This is the name of the network. Its unique ID.
+
+	var/list/network_devices 	= list()			// Devices utilizing the network.
+	var/list/mainframes 		= list()			// File servers serving files.
+	var/list/modems				= list()			// Modems capable of connecting to PLEXUS, the space internet.
+	var/list/broadcasters		= list()			// A list of anything broadcasting the good signal. Only one will be the /router.
+
+	var/obj/machinery/computer/exonet/mainframe/email_server	// A mainframe that's configured to be the email server. This doesn't take a special mainframe.
+	var/obj/machinery/computer/exonet/mainframe/log_server	// A mainframe that's the chosen main-log server.
+	var/obj/machinery/computer/exonet/mainframe/report_server// A mainframe that's the chosen main-report server.
+	var/obj/machinery/computer/exonet/broadcaster/router/router // The router that hosts the network. There can be only ONE!
+
+	var/list/administrators 	= list()			// A list of user_ids for administrators of the network.
+
+	var/default_domain								// OPTIONAL: If this is set, this is the default email domain for the exonet, allowing emails to be setup.
+	var/list/email_accounts 	= list()			// A list of emails configured for this exonet.
+
+	var/list/banned_nids		= list()			// A list of identification_ids on network devices that are banned from joining this network.
+	var/intrusion_detection_enabled					// LEGACY.
+	var/intrusion_detection_alarm					// LEGACY.
+
+
+/datum/exonet/New(var/new_ennid)
+	if(new_ennid)
+		ennid = new_ennid
+		GLOB.exonets[new_ennid] = src
+
+/datum/exonet/proc/change_ennid(var/new_ennid)
+	// aw god. pls never call this...
+	for(var/datum/network_device in network_devices)
+		if("ennid" in network_device.vars)
+			network_device.vars["ennid"] = new_ennid
```

Comment:
jesus on a stick
this is what extensions are for.
this var is for mapping etc right? Do you actually need to change it during runtime, instead of just using for initial extension setup and letting extension do the things?

Reply:
This is not for mapping. You can change the ennid on the router and that calls this.